routers/book-router.js

```diff
@@ -44,8 +44,12 @@ router.post("/", (req, res) => {
                     } else {
 			        	res.status(400).json({ error: "Book already exists" });
             }})
-		.catch(({ name, message, stack }) => res.status(500).json({ error: "Error, something went wrong", name, message, stack }));
-	} else {
+		.catch(({ name, message, stack }) => {
+            if (message.includes("Undefined binding(s) detected")) {
+                res.status(400).json({ error: "missing a necessary field", name, message, stack })    
+            } else { res.status(500).json({ error: "Error, something went wrong", name, message, stack }) };
```

Comment:
Please consider reformatting, spacing, and line breaks make this difficult to read.